rslearn/utils/vector_format.py

```diff
@@ -180,12 +182,81 @@ def from_config(name: str, config: dict[str, Any]) -> "TileVectorFormat":
         return TileVectorFormat(tile_size=config.get("tile_size", 512))
 
 
+class GeojsonCoordinateMode(Enum):
+    """The projection to use when writing GeoJSON file."""
+
+    # Write the features as is.
+    PIXEL = "pixel"
+
+    # Write the features in CRS coordinates (i.e., a projection with x_resolution=1 and
+    # y_resolution=1).
+    CRS = "crs"
+
+    # Write in WGS84 (longitude, latitude) coordinates.
+    WGS84 = "wgs84"
+
+
 @VectorFormats.register("geojson")
 class GeojsonVectorFormat(VectorFormat):
     """A vector format that uses one big GeoJSON."""
 
     fname = "data.geojson"
 
+    def __init__(
+        self, coordinate_mode: GeojsonCoordinateMode = GeojsonCoordinateMode.PIXEL
+    ):
+        """Create a new GeojsonVectorFormat.
+
+        Args:
+            coordinate_mode: the projection to use for coordinates written to the
+                GeoJSON files. PIXEL means we write them as is, CRS means we just undo
+                the resolution in the Projection so they are in CRS coordinates, and
+                WGS84 means we always write longitude/latitude. When using PIXEL, the
+                GeoJSON will not be readable by GIS tools since it relies on a custom
+                encoding.
+        """
+        self.coordinate_mode = coordinate_mode
+
+    def encode_to_file(
+        self, fname: UPath, projection: Projection, features: list[Feature]
+    ) -> None:
+        """Encode vector data to a specific file.
+
+        Args:
+            fname: the file to write to
+            projection: the projection of the raster data
+            features: the vector data
+        """
+        fc: dict[str, Any] = {"type": "FeatureCollection"}
+
+        # Identify target projection and convert features.
+        # Also set the target projection in the FeatureCollection.
+        # For PIXEL mode, we need to use a custom encoding so the resolution is stored.
+        output_projection: Projection
+        if self.coordinate_mode == GeojsonCoordinateMode.PIXEL:
+            output_projection = projection
+            fc["properties"] = projection.serialize()
+        else:
+            if self.coordinate_mode == GeojsonCoordinateMode.CRS:
+                output_projection = Projection(projection.crs, 1, 1)
+            elif self.coordinate_mode == GeojsonCoordinateMode.WGS84:
+                output_projection = WGS84_PROJECTION
+
+            fc["crs"] = {
+                "type": "name",
+                "properties": {
+                    "name": output_projection.crs.to_wkt(),
+                },
+            }
+
+        fc["features"] = []
+        for feat in features:
+            feat = feat.to_projection(output_projection)
+            fc["features"].append(feat.to_geojson())
+
+        with fname.open("w") as f:
```

Comment:
I think it is a good convention to log what we are writing and where when we write files. Makes it easier for someone using rslearn to understand what is happening